plugin/acl/acl.go

```diff
@@ -96,7 +99,19 @@ RulesCheckLoop:
 func matchWithPolicies(policies []policy, w dns.ResponseWriter, r *dns.Msg) action {
 	state := request.Request{W: w, Req: r}
 
-	ip := net.ParseIP(state.IP())
+	ip := []byte(nil)
+	if idx := strings.IndexByte(state.IP(), '%'); idx >= 0 {
+		ip = net.ParseIP(state.IP()[:idx])
+	} else {
+		ip = net.ParseIP(state.IP())
+	}
+
+	// if the parsing did not return a proper response then we simply return 'actionBlock' to
+	// block the query
+	if ip == nil {
+		log.Errorf("ParseIP failed for ip address: %v", state.IP())
```

Comment:
Please use the plugin logger  https://github.com/coredns/coredns/blob/master/plugin/pkg/log/plugin.go. It logs the plugin name along with the message.

example: https://github.com/coredns/coredns/blob/master/plugin/cache/setup.go#L17

A better error message would be "Unable to parse source address".